v3/lints/cabf_br/lint_invalid_subject_rdn_order.go

```diff
@@ -0,0 +1,145 @@
+/*
+ * ZLint Copyright 2024 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+/*
+ * Contributed by Adriano Santoni <adriano.santoni@staff.aruba.it>
+ * of ACTALIS S.p.A. (www.actalis.com).
+ */
+
+package cabf_br
+
+import (
+	"crypto/x509/pkix"
+	"encoding/asn1"
+
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/v3/lint"
+	"github.com/zmap/zlint/v3/util"
+)
+
+func init() {
+	lint.RegisterCertificateLint(&lint.CertificateLint{
+		LintMetadata: lint.LintMetadata{
+			Name:          "e_invalid_subject_rdn_order",
+			Description:   "Subject field attributes (RDNs) SHALL be encoded in a specific order",
+			Citation:      "BRs: 7.1.4.2",
+			Source:        lint.CABFBaselineRequirements,
+			EffectiveDate: util.CABFBRs_2_0_0_Date,
+		},
+		Lint: NewInvalidSubjectRDNOrder,
+	})
+}
+
+type invalidSubjectRDNOrder struct{}
+
+func NewInvalidSubjectRDNOrder() lint.LintInterface {
+	return &invalidSubjectRDNOrder{}
+}
+
+func (l *invalidSubjectRDNOrder) CheckApplies(c *x509.Certificate) bool {
+	return !util.IsCACert(c)
```

Comment:
I believe that this util is slightly more accurate by also excluding self signed certificates.

```suggestion
	return util.IsSubscriberCert(c)
```